tests/collections/collection.test.ts

```diff
@@ -845,6 +1128,102 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
       assert.ok(findOneRespDoc!.tags.length == 0);
       assert.ok(idsToCheck.has(findOneRespDoc!._id));
         });
+        it('should find & find doc $lt test', async () => {
+      interface Doc {
+        age: number;
+      }
+
+      const docs: Doc[] = Array.from({ length: 5 }, () => ({ age: 0 }));
+      docs.forEach((doc, index) => {
+          doc.age = doc.age + (index + 1);
+      });
+      const res = await collection.insertMany(docs);
+      assert.strictEqual(res.insertedCount, docs.length);
+      assert.strictEqual(res.acknowledged, true);
+      assert.strictEqual(Object.keys(res.insertedIds).length, 5);
+
+      const filter = { 'age': { '$lt': 3 } };
+      const expectedSet: Set<number> = new Set([1,2]);
+
+      const findRespDocs = await collection.find(filter).toArray();
+      assert.strictEqual(findRespDocs.length, 2);
+      findRespDocs.forEach((doc) => {
+          assert.ok(expectedSet.has(doc.age));
+      });
+        });
+        it('should find & find doc $lte test', async () => {
+      interface Doc {
+        age: number;
+      }
+
+      const docs: Doc[] = Array.from({ length: 5 }, () => ({ age: 0 }));
+      docs.forEach((doc, index) => {
+          doc.age = doc.age + (index + 1);
+      });
+      const res = await collection.insertMany(docs);
+      assert.strictEqual(res.insertedCount, docs.length);
+      assert.strictEqual(res.acknowledged, true);
+      assert.strictEqual(Object.keys(res.insertedIds).length, 5);
+
+      const filter = { 'age': { '$lte': 3 } };
+      const expectedSet: Set<number> = new Set([1,2,3]);
+
+      const findRespDocs = await collection.find(filter).toArray();
+      assert.strictEqual(findRespDocs.length, 3);
+      findRespDocs.forEach((doc) => {
+          assert.ok(expectedSet.has(doc.age));
+
+      });
+        });
+        it('should find & find doc $gt test', async () => {
+      interface Doc {
+        age: number;
+      }
+
+      const docs: Doc[] = Array.from({ length: 5 }, () => ({ age: 0 }));
+      docs.forEach((doc, index) => {
+          doc.age = doc.age + (index + 1);
+      });
+      const res = await collection.insertMany(docs);
+      assert.strictEqual(res.insertedCount, docs.length);
+      assert.strictEqual(res.acknowledged, true);
+      assert.strictEqual(Object.keys(res.insertedIds).length, 5);
+
+      const filter = { 'age': { '$gt': 3 } };
+      const expectedSet: Set<number> = new Set([4,5]);
+
+      const findRespDocs = await collection.find(filter).toArray();
+      assert.strictEqual(findRespDocs.length, 2);
+      findRespDocs.forEach((doc) => {
+          assert.ok(expectedSet.has(doc.age));
```

Comment:
This would still succeed even if the database returned 2 docs with `age: 4` or `age: 5`. I would recommend doing the following instead:

```
assert.deepStrictEqual(findRespDocs.map(doc => doc.age).sort(), [4, 5]);
```

The above will ensure that there are 2 documents, one with age: 4 and one with age: 5, in some order in the result.

Reply:
solved, thanks!